### ISSUE TITLE
Question on sphereface implementation 

### ISSUE THREAD

**Issue by lizhenstat** [ydwen/opensphere#19]
Hi, I have a question on sphereface implementation. 
In this SphereFace-Revived paper, the 4-th row in Table-1, the characteristic function delta(theta)=ita(theta)-psi(theta)
however in [the implementation](https://github.com/ydwen/opensphere/blob/main/model/head/sphereface.py#L37), it is as follows
```python
d_theta = phi_theta - cos_theta
```
which I am confused.

In the implementation of [SphereFaceR_H with "v2" type of magnitude](https://github.com/ydwen/opensphere/blob/main/model/head/spherefacer.py#L26).  The code follows the equation in Table-1

Any help would be appreciated and thanks for your time.

**Comment by ydwen** [comment]
Hi Lizhen,

Sorry for the confusion. These two deltas denote different terms in our loss function.

The delta in table 1 is to show the difference between the positive and negative logits.
While the delta in our implementation is used for a detach operator.
You could understand them as different terms.
